### DISCUSSION TITLE
Technical Problems Encountered

### DISCUSSION THREAD

**Discussion by mgilangjanuar** [mgilangjanuar/teledrive#5] [General]
This thread will be used for any issues we find in the development. Feel free to share your findings 🍻

**Comment by mgilangjanuar** [comment]
If you send too much code using [SendCode](https://core.telegram.org/method/auth.sendCode) you'll get 420. I'm only trying 4-5 times 🤣

<img width="561" alt="image" src="https://user-images.githubusercontent.com/3146378/131157768-b47a456b-1f2b-4822-8ec2-3e6049061305.png">

**Reply by Aqshola** [reply]
Damn :( , look like its telegram limitation for each number send 4/5 request auth
![Screenshot_20210828-060434_Chrome](https://user-images.githubusercontent.com/52363719/131196821-223ecb89-5419-4c12-9bef-3e640bea0625.jpg) soo maybe the solution is wait until the countdown finish so the user can send their auth code again

**Reply by mgilangjanuar** [reply]
Yup, actually we'll got the session key after the users sign in. So, the user doesn't need to re-login again for the future (except if the user terminate that session in the Telegram app).

We can use this as an authorization key to access the server side. And as a privacy concerns we'll not save it to anywhere except in the client side (eg. Cookie)

**Reply by Aqshola** [reply]
Woah i see, so the session will passed to client and stored to local storage *usually i  use this because of habit :( , just like treating jwt

**Reply by mgilangjanuar** [reply]
I think it will more secure and easy if the API response with header [set-cookie](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie) with same site attribute 🤔 

But, let's see later 🤣

**Comment by mgilangjanuar** [comment]
Nooo, Vercel has limited entity size in request body 😢 

<img width="432" alt="image" src="https://user-images.githubusercontent.com/3146378/133033479-fb256c04-d4d7-4513-a76b-bc2e4b49e8ae.png">

**Reply by mgilangjanuar** [reply]
https://vercel.com/support/articles/how-to-bypass-vercel-5mb-body-size-limit-serverless-functions

**Comment by mgilangjanuar** [comment]
Turns out the message ID in Telegram is different for each user. So, sharing files with other users is not as easy as I thought 😭 

<img width="588" alt="image" src="https://user-images.githubusercontent.com/3146378/133289612-41be0830-a8b7-4870-9b59-5744d31c9ee5.png">
<img width="560" alt="image" src="https://user-images.githubusercontent.com/3146378/133289684-ae0dd2f7-397e-466e-b7ad-10419d3f4a4f.png">

**Reply by mgilangjanuar** [reply]
Implemented workaround: just forward the message to other usernames 🤣

**Comment by bimg21** [comment]
Every time I try to creating an Telegram API key, it fails. It's persistent across devices, browsers and time. Anyone knows a solution?
![Screenshot from 2023-04-21 17-01-49](https://user-images.githubusercontent.com/124696075/233670092-09ee1d12-399a-429f-b358-95ac84307741.png)

**Reply by birdup000** [reply]
> Every time I try to creating an Telegram API key, it fails. It's persistent across devices, browsers and time. Anyone knows a solution? ![Screenshot from 2023-04-21 17-01-49](https://user-images.githubusercontent.com/124696075/233670092-09ee1d12-399a-429f-b358-95ac84307741.png)

Hi can you contact Telegram support https://telegram.org/support

**Reply by bimg21** [reply]
I did, about a week ago, no response.

**Reply by birdup000** [reply]
> I did, about a week ago, no response.

Please resolve this with Telegram as there's nothing we can do to have this fixed on your end. No Api Key = No Teledrive
